cs/HomeExercises/NumberValidatorTests.cs

```diff
@@ -7,6 +7,51 @@ namespace HomeExercises
 {
 	public class NumberValidatorTests
 	{
+		[TestCase(-1, 2, true, TestName = "precision < 0")]
+		[TestCase(0, 2, true, TestName = "precision = 0")]
+		[TestCase(1, -2, false, TestName = "scale < 0")]
+		[TestCase(1, 2, false, TestName = "precision < scale")]
+		[TestCase(1, 1, false, TestName = "precision = scale")]
+		public void Constructor_fail(int precision, int scale, bool onlyPositive)
+		{
+			Assert.Throws<ArgumentException>(() => new NumberValidator(precision, scale, onlyPositive));
+        }
+
+		[TestCase(1, 0, true, TestName = "precision > 0 and scale = 0")]
+		[TestCase(2, 1, true, TestName = "precision > scale")]
+        public void Constructor_success(int precision, int scale, bool onlyPositive)
+		{
+			Assert.DoesNotThrow(() => new NumberValidator(precision, scale, onlyPositive));
+        }
+
+		[TestCase(17, 2, true, "0.0", ExpectedResult = true, TestName = "value = 0.0")]
+		[TestCase(17, 2, true, "0", ExpectedResult = true, TestName = "value = 0")]
+		[TestCase(3, 2, true, "00.00", ExpectedResult = false, TestName = "value without point length > precision")]
+		[TestCase(3, 2, true, "-0.00", ExpectedResult = false, TestName = "minus counts per sign")]
+		[TestCase(3, 2, true, "+0.00", ExpectedResult = false, TestName = "plus counts as sign")]
+		[TestCase(3, 2, true, "+1.23", ExpectedResult = false, TestName = "value start with + with positive - scale = 1")]
+		[TestCase(17, 2, true, "0.000", ExpectedResult = false, TestName = "10")]
+		[TestCase(3, 2, true, "-1.23", ExpectedResult = false, TestName = "11")]
+		[TestCase(3, 2, true, "a.sd", ExpectedResult = false, TestName = "12")]
```

Comment:
Странные названия, навроде "12" ни о чём не говорят, название должно отражать какую-то часть спецификации. Пример хорошего названия - "Should fail when value contains letters instead of digits".